api/v1alpha2/contentlibraryitemimportrequest_types.go

```diff
@@ -0,0 +1,291 @@
+	// verification may fail and thus fail the import request.
+	SSLCertificate string `json:"sslCertificate,omitempty"`
+
+	// +optional
+
+	// Checksum contains the checksum algorithm and value calculated for the
+	// file specified in the URL. If omitted, the import request will not verify
+	// the checksum of the file.
+	Checksum *Checksum `json:"checksum,omitempty"`
+}
+
+// Checksum contains the checksum value and algorithm used to calculate that
+// value.
+type Checksum struct {
+	// +optional
+	// +kubebuilder:validation:Enum=SHA256;SHA512
+	// +kubebuilder:default=SHA256
+
+	// Algorithm is the algorithm used to calculate the checksum. Supported
+	// algorithms are "SHA256" and "SHA512". If omitted, "SHA256" will be used
+	// as the default algorithm.
+	Algorithm string `json:"algorithm"`
+
+	// Value is the checksum value calculated by the specified algorithm.
+	Value string `json:"value"`
+}
+
+// ContentLibraryItemImportRequestTargetItem contains the specification of the
+// target content library item for the import request.
+type ContentLibraryItemImportRequestTargetItem struct {
+	// +optional
+
+	// Name is the name of the new content library item that will be created
+	// in vSphere.
+	// If omitted, the content library item will be created with the same name
+	// as the name of the image specified in the spec.source.url in the
+	// specified library.
+	// If an item with the same name already exists in the specified library,
+	// the TargetValid condition will become false in the
+	// status.
+	Name string `json:"name,omitempty"`
+
+	// +optional
+
+	// Description is a description for a library item.
+	Description string `json:"description,omitempty"`
+
+	// +optional
+
+	// Type is the type of the new library item that will be created.
+	//
+	// The valid types depend on the type of underlying library:
+	// - LibraryType=ContentLibrary -- OVF
+	// - LibraryType=ContentLibrary -- ISO
+	//
+	// If omitted or the type is invalid, the TargetValid condition will be
+	// false.
+	//
+	// For the item type OVF, the default OVF security policy must be configured
+	// on the target library, otherwise the TargetValid condition will be false.
+	Type ContentLibraryItemType `json:"type,omitempty"`
+}
+
+// ContentLibraryItemImportRequestTarget is the target specification of an
+// import request.
+type ContentLibraryItemImportRequestTarget struct {
+	// +optional
+
+	// Item contains information about the library item to which the item will
+	// be imported in vSphere.
+	//
+	// If omitted, the library item will be created with the same name as the
+	// name of the image specified in the spec.source.url in the specified
+	// library.
+	//
+	// If an item with the same name already exists in the specified library,
+	// the TargetValid condition will be false.
+	Item ContentLibraryItemImportRequestTargetItem `json:"item,omitempty"`
+
+	// Library describes the name of the library in which the item will be
+	// created.
```

Comment:
Since we're adding a library type, it may be worthy to mention that this only applies to libraries of type ContentLibrary, we should add a validation rule in the webhook too.

Reply:
I think this is what the library's `AllowImport` field indicates. So we should have a webhook that prevents submitting import requests that reference libraries with `spec.allowImport: false`.